components/tidb_query/src/expr/builtin_string.rs

```diff
@@ -960,6 +960,28 @@ impl ScalarFunc {
         result.push(b'\'');
         Ok(Some(Cow::Owned(result)))
     }
+
+    // see https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_ord
+    #[inline]
+    pub fn ord<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<i64>> {
+        let s = try_opt!(self.children[0].eval_string_and_decode(ctx, row));
```

Comment:
According to copr-test, your function cannot work when string is an arbitrary binary string (instead of utf8).

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tikv_ghpr_integration-copr-test/detail/tikv_ghpr_integration-copr-test/1421/pipeline

Reply:
Yes, I have noticed that. But `std` doesn't include any function with can decode the string lazily (like `utf8.DecodeRuneInString` ).
I think I have to add `bstr` as a dependency.